deeplab/README.md

```diff
@@ -0,0 +1,202 @@
+# Semantic Segmentation in the Browser: DeepLab v3 Model
+
+## This model is a work-in-progress and has not been released yet. We will update this README when the model is released and usable
+
+This package contains a standalone implementation of the DeepLab inference pipeline, as well as a [demo](./demo), for running semantic segmentation using TensorFlow.js.
+
+![DeepLab Demo](./docs/deeplab-demo.gif)
+
+## Usage
+
+In the first step of semantic segmentation, an image is fed through a pre-trained model [based](https://github.com/tensorflow/models/blob/master/research/deeplab/g3doc/model_zoo.md) on MobileNet-v2. Three types of pre-trained weights are available, trained on [Pascal](http://host.robots.ox.ac.uk/pascal/VOC/voc2012/index.html), [Cityscapes](https://www.cityscapes-dataset.com) and [ADE20K](https://groups.csail.mit.edu/vision/datasets/ADE20K/) datasets.
+
+To get started, pick the model name from `pascal`, `cityscapes` and `ade20k`, and decide whether you want your model quantized to 1 or 2 bytes (set the `quantizationBytes` option to 4 if you want to disable quantization). Then, initialize the model as follows:
+
+```typescript
+import {load} from '@tensorflow-models/deeplab';
+const modelName = 'pascal';   // set to your preferred model, out of `pascal`, `cityscapes` and `ade20k`
+const quantizationBytes = 2; // either 1, 2 or 4
+const modelPromise = load({base: modelName, quantizationBytes});
+```
+
+By default, calling `load` initalizes the PASCAL variant of the model quantized to 2 bytes.
+
+If you would rather load custom weights, you can pass the URL in the config instead:
+
+```typescript
+import {load} from '@tensorflow-models/deeplab';
+const modelName = 'pascal';   // set to your preferred model, out of `pascal`, `cityscapes` and `ade20k`
+const quantizationBytes = 2; // either 1, 2 or 4
+const url = 'https://storage.googleapis.com/gsoc-tfjs/models/deeplab/quantized/1/pascal/model.json';
+const modelPromise = load({modelUrl: url});
+```
+
+This will initialize the model and return the promise of a `SemanticSegmentation` object.
+
+You can still pass the `base` attribute set to either `pascal`,`pascal`, `cityscapes` and `ade20k` in the argument to use the corresponding colormap and labelling scheme. Otherwise, you would have to provide those yourself.
```

Comment:
```suggestion
You can set the `base` argument to `pascal`, `cityscapes` or `ade20k` to use the corresponding colormap and labelling scheme. Otherwise, you would have to provide those yourself during segmentation.
```